router/core/src/main/scala/com/twitter/finagle/buoyant/DstTracing.scala

```diff
@@ -21,9 +21,11 @@ object DstTracing {
       private[this] val localDtabShow = dst.localDtab.show
       private[this] val pathShow = dst.path.show
       override def apply(conn: ClientConnection) = {
-        Trace.recordBinary("namer.dtab.base", baseDtabShow)
-        Trace.recordBinary("namer.dtab.local", localDtabShow)
-        Trace.recordBinary("namer.path", pathShow)
+        if (Trace.isActivelyTracing) {
```

Comment:
Yeah, the goal here to save on needless object construction.

For debug trace, I think we should make the debug tracer a full fledged tracer like:

```yaml
tracers:
- kind: io.l5d.debug
```

This solves this problem nicely so that isActivelyTracing will be true.

Note that isActivelyTracing returns true when:

1. at least one tracer is configured
2. and sampling is _not false_.